### PR TITLE
fix validation error (media only status)

### PR DESCRIPTION
**Pull request by MitarashiDango**
Fixed an issue that caused a validation error.  (fix #6672)

### CODE REVIEW

File: app/lib/activitypub/activity/create.rb [hunk 1]
```diff
@@ -21,9 +21,11 @@ def perform
 
   def process_status
     media_attachments = process_attachments
+    create_status_params = status_params
+    create_status_params['media_attachments'] = media_attachments
 
     ApplicationRecord.transaction do
-      @status = Status.create!(status_params)
+      @status = Status.create!(create_status_params)
```
Comment (ykzts) [inline app/lib/activitypub/activity/create.rb]:
```ruby
@status = Status.new(status_param)

process_tags(@status)
attach_media(@status, media_attachments)

@status.save!
```

🤔

cc @abcang

Comment (akihikodaki) [reply app/lib/activitypub/activity/create.rb]: That does not work, I think, because `attach_media` refers to the ID of the status.

Comment (ykzts) [reply app/lib/activitypub/activity/create.rb]: hm

Comment (Gargron) [reply app/lib/activitypub/activity/create.rb]: They can be reordered.. You can create media unattached first, and then pass to Status.create like in PostStatusService...

By the way, sorry I missed this..

Comment (abcang) [reply app/lib/activitypub/activity/create.rb]: I think that it is good to give media_attachments as a parameter of Status. And `attach_media` will not be used.

**Comment by abcang** [comment]
Probably, `OStatus::Activity::Creation` needs a similar fix.

**Review by akihikodaki — CHANGES_REQUESTED** [review]
Nice bug catch. Please see my review comment. Also, as @abcang said, don't forget to remove `attach_media`.

**Comment by MitarashiDango** [comment]
Addressed

**Review by akihikodaki — APPROVED** [review]
It looks good to me.

File: app/lib/activitypub/activity/create.rb [hunk 2]
```diff
@@ -54,6 +51,7 @@ def status_params
       visibility: visibility_from_audience,
       thread: replied_to_status,
       conversation: conversation_from_uri(@object['conversation']),
+      media_attachments: process_attachments,
```
Comment (abcang) [inline app/lib/activitypub/activity/create.rb]: In this case, it is not good to store the media in the transaction.

**Review by abcang — CHANGES_REQUESTED** [review]

File: app/lib/activitypub/activity/create.rb (hunk 2 above)
Comment (MitarashiDango) [reply app/lib/activitypub/activity/create.rb]: Save the return value of `process_status_params` to the `status_params` variable beforehand and pass it as an argument of `Status.create`?

Comment (abcang) [reply app/lib/activitypub/activity/create.rb]: It looks good!

**Comment by abcang** [comment]
It may be better to keep the function of limiting the number of media.

**Comment by MitarashiDango** [comment]
Addressed
- Fixed an issue where media was included in transaction.
- Restored the limit on the number of attached media.

**Review by abcang — APPROVED** [review]

**Review by Gargron — APPROVED** [review]

**Pull request merged**